core/trino-main/src/test/java/io/trino/operator/aggregation/AbstractTestAggregationFunction.java

```diff
@@ -184,6 +240,94 @@ public void testSlidingWindow()
         }
     }
 
+    @Test
+    public void testSlidingWindowForNaNAndInfinity()
+    {
+        if (this.getClass().getName().contains("TestDoubleSumAggregation") || this.getClass().getName().contains("TestDoubleAverageAggregation") || this.getClass().getName().contains("TestRealAverageAggregation")) {
```

Comment:
Instead of matching on the class names, I would move these tests specifically down into the specific class. If you need to share an implementation between the two for utility purposes, you can add it as a static method on (for example) `DoubleSumAggregation` and call it from both `DoubleSumAggregation` and `DoubleAverageAggregation`.

Reply:
Done